PacoMojica/ondemand-go-bootcamp
# First Deliverable

# What is in this PR?

Contains the requirements for the first deliverable, the API reads/writes a CSV file to generate the responses. I implemented error handlers where required and tried to use best practices and a clean architecture.

# How should this be manually tested?

Start the server by the default it should run on `http://localhost:8080/`, this can be changed in the file `config.yml`. The data of the app is stored in a CSV file called `db.csv`

The API has the following endpoints:

1. [GET] `/pokemon` returns a list of all the pokemon in the DB
2. [GET] `/pokemon/<ID>` returns the information of the pokemon that matches the ID,`null` otherwise
3. [POST] `create-pokemon` returns the data of the pokemon that was added to the DB

### create pokemon body

```json
{
    "id": 132,
    "name": "ditto",
    "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/132.png",
    "weight": 40,
    "height": 3,
    "base_experience": 0,
    "species": {
        "name": "ditto",
        "gender_rate": 0,
        "is_baby": false,
        "is_legendary": false,
        "is_mythical": false,
        "habitad": ""
    },
    "abilities": [
        {
            "name": "limber",
            "is_hidden": false
        },
        {
            "name": "imposter",
            "is_hidden": false
        }
    ],
    "moves": [
        "transform"
    ],
    "types": [
        "normal"
    ]
}
```
